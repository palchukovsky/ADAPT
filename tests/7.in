DECLARE testDecl;

SCOPE level1 {

  DECLARE testDecl;

  SCOPE level2 {

    DECLARE testDecl;

    SCOPE level3 { DECLARE testDecl; DECLARE level3decl; DECLARE lEvEl3decl; SCOPE level4{ DECLARE lEvEl4decl; ACCESS testDecl; ACCESS ::testDecl; } }

    DECLARE testDecl2;

    ACCESS testDecl2;
    ACCESS testDecl;
    ACCESS ::testDecl;
    ACCESS ::level1::testDecl;
    ACCESS level1::testDecl;
    ACCESS level3::level4::lEvEl4decl;

  }

  ACCESS level2::testDecl2;
    ACCESS testDecl;
  ACCESS ::level1::testDecl;
    ACCESS level1::testDecl;
  ACCESS level2::level3::level4::lEvEl4decl;

}

ACCESS level1::level2::testDecl2;
ACCESS testDecl;
ACCESS ::level1::testDecl;
ACCESS level1::testDecl;
ACCESS level1::level2::level3::level4::lEvEl4decl;

DECLARE lastDecl;
ACCESS lastDecl;
