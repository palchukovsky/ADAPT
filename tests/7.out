LINE 11 ACCESS ::level1::level2::level3::testDecl
LINE 11 ACCESS ::testDecl
LINE 15 ACCESS ::level1::level2::testDecl2
LINE 16 ACCESS ::level1::level2::testDecl
LINE 17 ACCESS ::testDecl
LINE 18 ACCESS ::level1::testDecl
LINE 19 ACCESS ::level1::testDecl
LINE 20 ACCESS ::level1::level2::level3::level4::lEvEl4decl
LINE 24 ACCESS ::level1::level2::testDecl2
LINE 25 ACCESS ::level1::testDecl
LINE 26 ACCESS ::level1::testDecl
LINE 27 ACCESS ::level1::testDecl
LINE 28 ACCESS ::level1::level2::level3::level4::lEvEl4decl
LINE 32 ACCESS ::level1::level2::testDecl2
LINE 33 ACCESS ::testDecl
LINE 34 ACCESS ::level1::testDecl
LINE 35 ACCESS ::level1::testDecl
LINE 36 ACCESS ::level1::level2::level3::level4::lEvEl4decl
LINE 39 ACCESS ::lastDecl
